### ISSUE TITLE
Include CI tools

### ISSUE THREAD

**Issue by uilianries** [datacom-teracom/l2tester#3]
Hi!

Thanks to push this great project into Github.

I would like to include some CI tools (e.g. Travis, Circle CI) to validate code warnings and compiler errors.

WDYT?

**Comment by e-witter** [comment]
Great, I'll do it soon.

**Comment by evedovelli** [comment]
No problem at all. Indeed, it is a desired enhancement.

**Comment by henriquenj** [comment]
IMHO we should aim to test within a Ubuntu 14.04 VM first (which travis provide it for us) since this OS is the one we know it works. We can then gradually add more jobs for newer versions of Ubuntu and other distros as a second step.

**Comment by uilianries** [comment]
I use a large number of docker images to distribute Conan packages, including clang and gcc. All images are based on Ubuntu, but only few images are based on 14.04.  
Also, there are official images for gcc and clang on hub.docker  
I'm just saying that is not necessary to create a new docker image to build the project.

**Comment by henriquenj** [comment]
I'm not suggesting we generate our own docker image, or any image. I'm suggesting that we use Ubuntu 14.04 as a starting point, using the standard image provided by Travis.

**Comment by uilianries** [comment]
The problem running on standard Travis image is that you are not "able" to reproduce the same environment on your side...  So I commented about docker because the same docker image could be executed on both sides. But maybe you know some official image provided by Travis CI with cpp config on [docker hub](https://hub.docker.com/r/travisci).

**Comment by henriquenj** [comment]
Fair point. I was indeed assuming I could download a standard Travis image and then just run it on my machine. If this is not true it will be a disadvantage of using Travis.

**Comment by uilianries** [comment]
I found some tutorials running Travis over Docker, but I didn't try. Anyway, Travis might be only the "engine" to run all jobs. Run docker over Travis is pretty easy: https://docs.travis-ci.com/user/docker/

**Comment by henriquenj** [comment]
Nice finding, thanks. 

I'm not very concerned about using standard images just to compile l2tester (which I think it's the natural first step). Even if we can't run these images on a local machine, l2tester does not have a super elaborated build process. I'm _more_ concerned when we attempt to run automated test, which I guess will involve some low level Linux manipulation, such as creating virtual Ethernet interfaces.
